src/Build/BackEnd/Shared/BuildResult.cs

```diff
@@ -100,6 +112,24 @@ public class BuildResult : INodePacket, IBuildResults
         /// </summary>
         private Dictionary<string, string> _savedEnvironmentVariables;
 
+        /// <summary>
+        /// Lock object for the dictionary <see cref="_savedEnvironmentVariables"/>.
+        /// </summary>
+        private readonly object _lock = new object();
+
+        /// <summary>
+        /// When this key is in the dictionary <see cref="_savedEnvironmentVariables"/>, serialize the build result version.
+        /// </summary>
+        private const string VersionKeyName = "MSBUILDFEATUREBUILDRESULTHASVERSION";
```

Comment:
Oh, great idea! This will decrease the probability of collision to zero indeed.

Reply:
I cannot do this in the implementation with the lock - because the read of the dictionary is not protected by the lock it may lead to a crash trying to set this variable.